### ISSUE TITLE
Consider adding a Form Extension for disabling csrf when the class is used

### ISSUE THREAD

**Issue by tarjei** [dunglas/DunglasAngularCsrfBundle#2]
Something like this:

```php
/**
 * Class DisableCSRFExtension
 *
 * @author Grégoire Pineau
 * @author Tarjei Huse (Extended the class to disable csrf on forms)
 * 
 */
class DisableCSRFExtension extends AbstractTypeExtension
{
    private $securityContext;
    /**
     * @var RequestStack
     */
    private $requestStack;
    /**
     * @var AngularCsrfTokenManager
     */
    private $angularCsrfTokenManager;
    /**
     * @var RouteMatcherInterface
     */
    private $routeMatcher;
    /**
     * @var array
     */
    private $routes;
    /**
     * @var
     */
    private $headerName;


    /**
     * @param SecurityContextInterface $securityContext
     * @param RequestStack             $requestStack
     * @param AngularCsrfTokenManager  $angularCsrfTokenManager
     * @param RouteMatcherInterface    $routeMatcher
     */
    public function __construct(SecurityContextInterface $securityContext,
                                RequestStack $requestStack,
                                AngularCsrfTokenManager $angularCsrfTokenManager,
                                RouteMatcherInterface $routeMatcher,
                                array $routes,
                                $headerName
    )
    {
        $this->securityContext = $securityContext;
        $this->requestStack = $requestStack;
        $this->angularCsrfTokenManager = $angularCsrfTokenManager;
        $this->routeMatcher = $routeMatcher;
        $this->routes = $routes;
        $this->headerName = $headerName;
    }

    public function setDefaultOptions(OptionsResolverInterface $resolver)
    {
        if (!$this->securityContext->getToken()) {
            return;
        }
        $request = $this->requestStack->getCurrentRequest();
        if (!$this->routeMatcher->match($request, $this->routes)
        ) {
            return;
        }
        $value = $request->headers->get($this->headerName);
        
        if ($this->angularCsrfTokenManager->isTokenValid($value)) {
            $resolver->setDefaults(array(
                'csrf_protection' => false,
            ));
        }
    }

    public function getExtendedType()
    {
        return 'form';
    }
}
```

**Comment by xavierleune** [comment]
:+1:

**Comment by dunglas** [comment]
PRs are welcome!

**Comment by dunglas** [comment]
Closed by #9.
